### ISSUE TITLE
Schema validation doesn't differentiate between `int64` and `Int64` dtypes

### ISSUE THREAD

**Issue by tamargrey** [alteryx/woodwork#658]
Our definition of what's a valid schema for a DataFrame involves that the dtypes of the DataFrame all match the valid dtype for that column's LogicalType. This means that an `int64` dtype DataFrame series should be invalid for a column with logical type `Integer`.

However, we do not catch that case currently:

![image.png](https://images.zenhubusercontent.com/5f15e80b8f74aa3e95d9e419/24b12a7f-6ed6-4418-9fff-b98748bd46b5) 

This is how we attempt to catch that case:
```python
        df_dtype = dataframe[name].dtype
        valid_dtype = _get_valid_dtype(dataframe[name], schema.logical_types[name])

        if df_dtype != valid_dtype:
            ....
```
This is probably related to how pandas differentiates between the nullable and non-nullable dtypes, so we need to make sure we check for valid dtypes (here and across Woodwork) in ways that do differentiate between nullable and non-nullable dtypes.

**Comment by thehomebrewnerd** [comment]
Interesting. Wonder if this is a bug in pandas or if this is intended behavior? 

Can we check the string representations of the dtypes here instead?

**Comment by tamargrey** [comment]
Checking the string representation does work! (it also helps in other ways because, interestingly, Pandas doesn't let you compare `'object'` dtypes to `'string'` dtypes so that was raising unexpected errors if an expected string column had been converted to the object dtype at some point. 

I think this is a fix that can go into #653, but I would like to do a pass through the rest of the code and figure out where else we might have a similar issue. Plus I'd like to really understand what's going on here with Pandas (I feel like this is probably their intended behavior and it's just that we care more than they do about differentiating `int64` from `Int64`) and make sure that we're properly testing this.

**Comment by tamargrey** [comment]
Fixed by #681
